aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::aggregator_extension::AggregatorID;
+
+/// Returns a value of an aggregator from cache or global storage.
+///   - Ok(..)       if aggregator value exists
+///   - Err(..)      otherwise.
+pub trait AggregatorResolver {
```

Comment:
Let's either add another method for the "cheap" read, e.g. approximate_aggregator_value, or add a parameter to resolve_aggregator_value (allow_approximate), or we can find a better word than approximate to capture the "cheap" / "predictive" nature.

Should we bundle grabbing a new ID API into the same trait? @igor-aptos @vusirikala @zekun000 I think it makes sense to have a trait for all aggregator related connections to the adapter?

Reply:
@gelash should we add this for main or aggregator_v2?

I think having a trait for adapter makes perfect sense, we can land things on main since these APIs can be used for V1 (apart from cheap read).

Having it combined with GUID sounds good.